src/uniswapV3/UniswapV3Wrapper.sol

```diff
@@ -112,10 +112,11 @@ contract UniswapV3Wrapper is BaseWrapper, IUniswapV3FlashCallback {
         require(msg.sender == address(_activePool), "UniswapV3Wrapper: Only active pool");
 
         uint256 fee = fee0 > 0 ? fee0 : fee1;
-        IERC20 asset = IERC20(fee0 > 0 ? IUniswapV3Pool(msg.sender).token0() : IUniswapV3Pool(msg.sender).token1());
-        uint256 amount = asset.balanceOf(address(this));
+        IERC20 asset = IERC20(fee0 > 0 ? IUniswapV3Pool(msg.sender).token0() : IUniswapV3Pool(msg.sender).token1()); // TODO: Wouldn't it be cheaper to pass this on `params`?
```

Comment:
I thought on having amount and asset on params, but wasn't sure if worth the complexity

Reply:
It would be cheaper on params than a CALL and a SLOAD, I assume

Reply:
Yes ofc

Reply:
But other than on mainnet, the difference should be negligible. 
If we're optimising for that, we should even get rid of the sstore to save the current Pool, there're other ways to verify the caller, that should make by far the biggest difference.

Reply:
Fair enough, it was fun to code, but we can drop it. Let's leave just the uniswap fix.